### PR TITLE
docs(cookbook): Add a submenu extension "cookbook" example

### PR DESCRIPTION
**Pull request by bradh**

### CODE REVIEW

File: docs/cookbook/submenu/src/cookbook-submenu.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      handler: plugin.cookbook_submenu.handleItem
+    });
+  }
+
+};
+
+
+/**
+ * If our example group should be shown.
+ * @param {Object|undefined} context The menu context.
```
[the remark this answers is not in the document]
Comment (bradh) [reply docs/cookbook/submenu/src/cookbook-submenu.js]: Just to be clear, you're just asking for adding `@this {os.ui.menu.MenuItem}` to the applicable functions?

[the remark this answers is not in the document]
Comment (wallw-bits) [reply docs/cookbook/submenu/src/cookbook-submenu.js]: Correct.

File: docs/cookbook/submenu/src/cookbook-submenu.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    var group = root.find(plugin.cookbook_submenu.MYGROUP);
+    if (!group) {
+      group = root.addChild({
+        type: os.ui.menu.MenuItemType.GROUP,
+        label: plugin.cookbook_submenu.MYGROUP,
+        tooltip: 'Added by cookbook submenu example',
+        beforeRender: plugin.cookbook_submenu.shouldShowGroup_
+      });
+    }
+    var submenu1 = group.addChild({
```
[the remark this answers is not in the document]
Comment (bradh) [reply docs/cookbook/submenu/src/cookbook-submenu.js]: Added as follow-up commit.

File: docs/cookbook/submenu/src/cookbook-submenu.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bradh) [reply docs/cookbook/submenu/src/cookbook-submenu.js]: Fixed in follow-up commit.

**Pull request merged**